### PR TITLE
Remove utility-types and other TS cleanup

### PR DESCRIPTION
**Pull request by jtbandes**
Some cleanup around types of hooks.ts and related code. Remove utility-types.

### CODE REVIEW

File: app/PanelAPI/useMessageReducer.ts [hunk 1]
```diff
@@ -135,12 +135,12 @@ type Props<T> = {
   // TODO(steel): Eventually we should deprecate these multiple ways of getting data, and we should
   // always have blocks available. Then `useMessageReducer` should just become a wrapper around
   // `useBlocksByTopic` for backwards compatibility.
-  preloadingFallback?: boolean | null | undefined;
+  preloadingFallback?: boolean | undefined;
```
Comment (defunctzombie) [inline app/PanelAPI/useMessageReducer.ts:138]: `| undefined`  implied by the `?` ?

File: app/PanelAPI/useMessageReducer.ts [hunk 2]
```diff
@@ -116,15 +116,15 @@ function useSubscriptions({
...
 
 type Props<T> = {
-  topics: ReadonlyArray<RequestedTopic>;
+  topics: readonly RequestedTopic[];
 
   // Functions called when the reducers change and for each newly received message.
   // The object is assumed to be immutable, so in order to trigger a re-render, the reducers must
   // return a new object.
-  restore: (arg0: T | null | undefined) => T;
+  restore: (arg0: T | undefined) => T;
```
Comment (defunctzombie) [inline app/PanelAPI/useMessageReducer.ts:127]: `arg0?: T`

File: app/components/TimeBasedChart/HoverBar.tsx [hunk 3]
```diff
@@ -35,7 +35,7 @@ type Props = {
   componentId: string;
   // We don't need to (and shouldn't) rerender when the scale-bounds changes under the cursor -- the
   // bar should stay under the mouse. Only rerender when the mouse moves (using useSelector).
-  scaleBounds: { current: ReadonlyArray<ScaleBounds> | null | undefined };
+  scaleBounds: { current: readonly ScaleBounds[] | null | undefined };
```
Comment (defunctzombie) [inline app/components/TimeBasedChart/HoverBar.tsx:38]: `current?:` ? Or is this trying to pretend to be a ref?

File: app/dataProviders/types.ts [hunk 4]
```diff
@@ -57,17 +57,17 @@ import { RosDatatypes } from "@foxglove-studio/app/types/RosDatatypes";
 // MeasureDataProvider.
 
 export type GetMessagesTopics = Readonly<{
-  parsedMessages?: ReadonlyArray<string> | null | undefined;
-  rosBinaryMessages?: ReadonlyArray<string> | null | undefined;
-  bobjects?: ReadonlyArray<string> | null | undefined;
+  parsedMessages?: readonly string[] | null | undefined;
```
Comment (defunctzombie) [inline app/dataProviders/types.ts:60]: `| undefined` are not needed here.

File: app/panels/ThreeDimensionalViz/Interactions/ObjectDetails.tsx [hunk 5]
```diff
@@ -46,19 +45,11 @@ const SObjectDetails = styled.div`
   padding: 12px 0 16px 0;
 `;
 
-type CommonProps = $ReadOnly<{ interactionData: InteractionData | null | undefined }>;
+type CommonProps = { readonly interactionData: InteractionData | null | undefined };
```
Comment (defunctzombie) [inline app/panels/ThreeDimensionalViz/Interactions/ObjectDetails.tsx:48]: `interactionData?:`

**Comment by defunctzombie** [comment]
:shipit:

File: app/panels/ThreeDimensionalViz/Interactions/ObjectDetails.tsx (hunk 5 above)
Comment (jtbandes) [reply app/panels/ThreeDimensionalViz/Interactions/ObjectDetails.tsx:48]: Will do these in a separate PR.

**Pull request merged**